src/Microsoft.AspNetCore.Authorization.Middleware/AuthorizationAppBuilderExtensions.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Authorization;
+
+namespace Microsoft.AspNetCore.Builder
+{
+    public static class AuthorizationAppBuilderExtensions
+    {
+        public static IApplicationBuilder UseAuthorization(this IApplicationBuilder app)
+        {
+            if (app == null)
+            {
+                throw new ArgumentNullException(nameof(app));
+            }
+
+            return app.UseMiddleware<AuthorizationMiddleware>();
```

Comment:
is there any sort of options object that should be passed in? or is it always one from DI?

Reply:
IAuthorizationPolicyProvider currently always comes from DI

Reply:
Is there a need to be able to add multiple instances of this? E.g. `UseAuthorization("policyA")`? It might be useful for that StaticFile example.

Reply:
Yeah, that's the general extensibility point for if you want dynamic policies for example.

Reply:
Yeah I asked about what the intended usage of this middleware is, and rather than it being analogous to a single `[Authorize(policyName)]` invocation, its intended to be a single effective authorization check that uses the endpoint to compute the effective policy for the entire request.

Reply:
I'm assuming if multiple UseAuthorization() end up in the pipeline, they would become no-ops similar to [Authorize] attributes in MVC right?

Reply:
I was referring to branched pipelines.

Reply:
> Yeah I asked about what the intended usage of this middleware is, and rather than it being analogous to a single `[Authorize(policyName)]` invocation, its intended to be a single effective authorization check that uses the endpoint to compute the effective policy for the entire request.

Why not both? Allowing us to specify a policy to run would help extend AuthZ to non-endpoint aware middleware.

Reply:
I can see `UseAuthorization("policyA")` having value. Will add.